src/Our.Umbraco.RobotsTxtEditor/Controllers/RobotsTxtEditorApiController.cs

```diff
@@ -32,24 +35,69 @@ public RobotsTxtEditorModel GetRobotsText()
                 return new RobotsTxtEditorModel
                 {
                     FileExists = true,
-                    FileContents = reader.ReadToEnd()
+                    FileContents = reader.ReadToEnd().TrimEnd()
                 };
             }
         }
 
         [HttpPost]
-        public bool SaveRobotsText(RobotsTxtEditorModel vm)
+        public RobotsTxtResponseModel SaveRobotsText(RobotsTxtEditorModel vm)
         {
             //do something that would save it here
             var filePath = IOHelper.MapPath("~/robots.txt");
 
-            //create the file and then write to it
-            using (var sw = File.CreateText(filePath))
-            {
-                sw.WriteLine(vm.FileContents);
-            }
+            var contents = vm.FileContents;
+
+            var errorMessages = ValidateRobotsTxt(contents);
+
+            if (errorMessages.Count == 0)
+            {
+                //create the file and then write to it
+                using (var sw = File.CreateText(filePath))
+                {
+                    sw.WriteLine(contents);
+                }
+            }
+
+            return new RobotsTxtResponseModel
+            {
+                ErrorMessages = errorMessages,
+                Success = errorMessages.Count == 0
+            };
+        }
 
-            return true;
+        /// <summary>
+		/// A very basic Robots.txt validation.
+		/// </summary>
+		/// <param name="contents">The contents of the Robots.txt file.</param>
+		/// <returns>A list of Pair objects, containing the line number and detail of the error.</returns>
+		public List<Pair> ValidateRobotsTxt(string contents)

```

Comment:
Teeny nitpick - mix of tabs & spaces.
(Personally, I'm a tabs fan, but compromised over the years ... mostly due to VS IDE ships with spaces)
We could add an `.editorconfig` to the root to ensure whitespace indenting (and other code formatting options)?